src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImportPowerShellDataFile.cs

```diff
@@ -0,0 +1,122 @@
+--********************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+using System.Management.Automation.Language;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Import-PowerShellDataFile command.
+    /// </summary>
+    [Cmdlet(VerbsData.Import, "PowerShellDataFile", DefaultParameterSetName = "Path",
+        HelpUri = "https://go.microsoft.com/fwlink/?LinkID=623621", RemotingCapability = RemotingCapability.None)]
+    public class ImportPowerShellDataFileCommand : PSCmdlet
+    {
+        bool _isLiteralPath;
+
+        /// <summary>
+        /// Path specified, using globbing to resolve
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = "Path")]
+        [ValidateNotNullOrEmpty]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public string[] Path { get; set; }
+
+        /// <summary>
+        /// Specifies a path to one or more locations, without globbing
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = "LiteralPath")]
+        [ValidateNotNullOrEmpty]
+        [Alias("PSPath")]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public string[] LiteralPath
+        {
+            get { return Path; }
+            set { _isLiteralPath = true; Path = value; }
+        }
+
+        /// <summary>
+        /// For each path, resolve it, parse it and write all hashtables
+        /// to the output stream
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            foreach (var path in ResolveFilePaths(Path, _isLiteralPath))
+            {
+                Token[] tokens;
+                ParseError[] errors;
+                var ast = Parser.ParseFile(path, out tokens, out errors);
+                if (errors.Length > 0)
+                {
+                    WriteError(BuildInvalidDataFileErrorRecord(path, "CouldNotParseAsPowerShellDataFile"));
+                }
+                else
+                {
+                    var data = ast.Find(a => a is HashtableAst, false);
+                    if (data != null)
+                    {
+                        WriteObject(data.SafeGetValue());
+                    }
+                    else
+                    {
+                        WriteError(BuildInvalidDataFileErrorRecord(path, "CouldNotParseAsPowerShellDataFileNoHashtableRoot"));
+                    }
+                }
+            }
+        }
+
+        private List<string> ResolveFilePaths(string[] filePaths, bool isLiteralPath)
```

Comment:
This method is copied from https://github.com/PowerShell/PowerShell/blob/9a195f44c17773fdbb5a2e6a09de0c880072df23/src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs#L1723

Can we avoid such copy-paste?